src/auth/src/retry.rs

```diff
@@ -103,31 +152,90 @@ where
             }
         };
 
-        retry_loop(
+        // The on_retry callback sends intermediate errors and stores the last delay.
+        let last_delay = self.last_delay.clone();
+        let tx_for_retry = self.token_watch_tx.clone();
+        let on_retry = move |_, error: &gax::error::Error, delay: Duration| {
```

Comment:
@coryan the API change is in `retry_loop_internal.rs`.  The updated API is used here.

Reply:
Ack. What I was suggesting is to create a wrapper in line 180 here, so instead of :

```rust
    self.retry_policy.clone(),
```

You would write:

```
    Arc::new(CaptureRetry::new(self.retry_policy.clone())),
```

where `CaptureRetry` is something like:


```rust
struct CaptureRetry {
  inner: Arc<dyn RetryPolicy>
  stuff: ...
}

impl CaptureRetry {
  fn new(inner: Arc<dyn RetryPolicy>) -> Self { Self { inner, ... } }
}

impl RetryPolicy for CaptureRetry {
  fn on_error(...) {
      // the thing you do in the new callback
  }
}
```

But if you really need something ad-hoc: that is an indication that the retry loop for auth is sufficiently different that maybe we do not want to reuse it.

Reply:
Thank you @coryan . Yes creating a private wrapper like this makes sense. But I believe we would have to do the wrapper on backoff policy and not retry policy (as explained [here](https://github.com/googleapis/google-cloud-rust/pull/2833#discussion_r2264124013)).

If later we decide that we need info from retry policy as well, we could have wrapper on both retry policy and backoff policy and compose the information for use in auth lib.

If this makes sense, we can close these PRs for now and reopen when we work on them since the solution described here is not changing any public API surface.

Reply:
> Thank you @coryan . Yes creating a private wrapper like this makes sense. But I believe we would have to do the wrapper on backoff policy and not retry policy

SGTM. Or even the `sleep` function.... Either way, does not look like you need a custom callback?

 
> If this makes sense, we can close these PRs for now and reopen when we work on them since the solution described here is not changing any public API surface.

What about the changes to `CredentialsError`?